MapboxCoreNavigation/RouteController.swift

```diff
@@ -292,11 +288,10 @@ open class RouteController: NSObject {
         let averageRelativeAngle = (relativeAnglepointBehind + relativeAnglepointAhead) / 2
         let absoluteDirection = wrap(wrappedCourse + averageRelativeAngle, min: 0 , max: 360)
         
-        // If the course is inaccurate and the user is on the route,
-        // calculate a rough estimate as to what the course should be at that point on the route.
-        if location.course <= 0 && snappedCoordinate.distance < RouteControllerUserLocationSnappingDistance {
-            let calculatedCourse = wrap((wrappedPointBehind + wrappedPointAhead) / 2, min: 0 , max: 360)
-            return CLLocation(coordinate: snappedCoordinate.coordinate, altitude: location.altitude, horizontalAccuracy: location.horizontalAccuracy, verticalAccuracy: location.verticalAccuracy, course: calculatedCourse, speed: location.speed, timestamp: location.timestamp)
+        // If the course is inaccurate or the speed is low and the user is on the route,
+        // snap the users location and course since we know the snapped location and course is more accurate.
+        if location.course <= 0 || location.speed <= 2, snappedCoordinate.distance < RouteControllerUserLocationSnappingDistance {
```

Comment:
Factor out 2 km/h into a constant. Also, for reference, human walking speed is between 4 and 7 km/h.

Reply:
That's some fast walking.